### ISSUE TITLE
use native arrays crash with v3 tiles

### ISSUE THREAD

**Issue by GISnick** [unitycoder/UnityPointCloudViewer#143]
In the latest version of the point cloud reader, i ran the Tiles-viewerV3 demo scene and unity editor crashes on loading of the point cloud if i enable **_use native arrays_**
If i do not use this option, everything works as expected. 

I will provide more info if you are unable to reproduce the issue.

Point cloud viewer and tools
version: 2.80
Latest release date: Oct 11, 2023

Unity 2022.3.12f1
URP

custom point cloud converted from your converter that works in the previous version ok with native arrays

piece of editor.log
```


=================================================================
	Native Crash Reporting
=================================================================
Got a UNKNOWN while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Managed Stacktrace:
=================================================================
	  at <unknown> <0xffffffff>
	  at Unity.Collections.LowLevel.Unsafe.UnsafeUtility:MemCpy <0x00097>
	  at PointCloudHelpers.PointCloudMath:MoveFromByteArray <0x001f2>
	  at unitycodercom_PointCloudBinaryViewer.PointCloudViewerTilesDX11:ReadPointCloudThreaded <0x003a2>
	  at unitycodercom_PointCloudBinaryViewer.PointCloudViewerTilesDX11:LoaderWorkerThread <0x00162>
	  at System.Threading.ThreadHelper:ThreadStart_Context <0x000e1>
	  at System.Threading.ExecutionContext:RunInternal <0x001b5>
	  at System.Threading.ExecutionContext:Run <0x0002a>
	  at System.Threading.ExecutionContext:Run <0x0005a>
	  at System.Threading.ThreadHelper:ThreadStart <0x0004a>
	  at <Module>:runtime_invoke_void__this___object <0x000c9>

```

**Comment by unitycoder** [comment]
ok i'll check!

**Comment by unitycoder** [comment]
@GISnick didnt seemto happen on 2019.4.19f1 or 2023.3.0a13, 2022.3.7f1 (birp),
going to test URP next, but if you happen to have screenshot of your V3 viewer settings, i can check those also. (in case anything is different)

**Comment by GISnick** [comment]
will do thanks! so on further inspection, it happens only when i use:

- native arrays / true
- release tile memory  / true
- limit tile points  / true

native arrays works if  _limit tile points = false_, and limit tile points works if _native arrays = false._ 

but together they crash the editor

this is a very large cloud. ill do some messing around, maybe generate a new cloud with the newer converter and see if i can reproduce and give you a small example. may have to do with my massive dense cloud 
![image](https://github.com/unitycoder/UnityPointCloudViewer/assets/119757748/432c432b-d8cb-4627-814d-044162d00162)

**Comment by unitycoder** [comment]
Ah ok, must be related to that setting then, no need to send data i'll test those first.

**Comment by unitycoder** [comment]
ok simple bug, i'll upload the fix in asset store in few days or so.. but you can fix it at:
```
// open PointCloudMath.cs
// find line
UnsafeUtility.MemCpy(&dstAddr[0], &srcAddr[0], src.Length);

// replace that 1 line with these 2 lines
long size = src.Length < dst.Length ? src.Length : dst.Length;
UnsafeUtility.MemCpy(&dstAddr[0], &srcAddr[0], size);
```

**Comment by GISnick** [comment]
yes sir!! thanks working great now

**Comment by unitycoder** [comment]
fixed in v2.90 asset store version also.
